packages/react-components/react-table/src/components/DataGrid/useDataGrid.ts

```diff
@@ -45,11 +50,24 @@ export const useDataGrid_unstable = (props: DataGridProps, ref: React.Ref<HTMLEl
       onSelectionChange,
       selectionMode: selectionMode ?? 'multiselect',
     }),
+    useTableColumnSizing_unstable({
+      onColumnResize,
+      columnSizingOptions,
+      // The selection cell is not part of the columns, therefore its width needs to be subtracted
+      // from the container to make sure the columns don't overflow the table.
+      containerWidthOffset: selectionMode ? -CELL_WIDTH : 0,
```

Comment:
This seems to be a simple solution for the problem, and also most flexible one, as with Table, users can just render anything in front of the real columns, so this feels like a better solution than just adapting logic for selection